File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.java


import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.LinkedList;
import java.util.Queue;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());

        Queue<Integer> q = new LinkedList<>();

        for(int i=1;i<=n;i++){
            q.offer(i);
        }

        System.out.print("<");
        for(int i=0;i<n;i++){
            int count = 0;
            while(!q.isEmpty()){
                int temp = q.poll();
                count++;

                if(count == k){
                    System.out.print(temp);
                    break;
                }

                q.offer(temp);
            }

            if(i!=n-1) System.out.print(", ");

        }
        System.out.print(">");
    }
}

File: 백준/Silver/11866. 요세푸스 문제 0/요세푸스 문제 0.cc
#include <iostream> 
#include <queue>
using namespace std;
 
int N,K;
queue<int>q;

int main(void)
{
        cin >> N >> K;
        
        for(int i=1;i<=N;i++) q.push(i);

        cout << "<";

        for(int i=0;i<N;i++){
            int count = 0;

            while(q.size() != 0){
                count++;
                if(count == K) break;

                q.push(q.front());
                q.pop();
            }

            cout << q.front();
            if(q.size() != 1) cout << ", ";
            q.pop();
        }
        cout << ">";
        
        return 0;
}